### DISCUSSION TITLE
What can lead to such deviations of camera view?

### DISCUSSION THREAD

**Discussion by tiloc** [ramokz/phantom-camera#292] [Question / Help Wanted]
I am trying to migrate from my hand-rolled camera setup based on Camera3D to Phantom Camera. In my hand-rolled setup I have a Marker3D 1.8m behind my characters left shoulder and another Marker3D 2m in front of their face. In my camera's _physics_process I moved the camera to the position of the shoulder marker and made it look at the marker in front of the face. This gave the impression of a camera man running behind my character (albeit with no dampening).

Now with Phantom Camera I have created the following setup:
<img width="1570" alt="image" src="https://github.com/ramokz/phantom-camera/assets/16414047/3d211c85-c0af-4773-8076-868b8e70032b">

I let the camera follow my characters center at eye level with an offset to place it roughly behind the left shoulder. I make it look at the center at eye level with an offset 2m in front of the characters. In the preview, everything looks as desired. There are no scripts associated with the camera, only my character is scripted and the CenterEyeLevel marker is moving around with it.

Now as soon as I actually run my game, the camera shows a completely different perspective (and I haven't figured out yet, which one exactly - I believe it is looking away from the character to behind the character)
<img width="958" alt="image" src="https://github.com/ramokz/phantom-camera/assets/16414047/0b699071-38aa-4a1a-a085-9996997bd113">

Any ideas how the preview and the actual outcome could be so different?

**Comment by ramokz** [comment]
Difficult to say without a scene to test in.

But the question has appeared before, where the reason is typically due to the `Collision Mask` property clashing with another — such as character's.

**Reply by tiloc** [reply]
I have disabled all collision masks, with no difference in result. I have played around with it some more, and it seems the camera during gameplay is turned 180° and looking away from the player.

**Reply by ramokz** [reply]
Doing some more tests, it's likely due to using both `Third Person` `follow mode` and a `Look At` option.
The two systems haven't been made to be entirely compatible yet, as the way `Third Person` works in the editor differs from how it works during runtime.

**Reply by tiloc** [reply]
Ok, yes, I was using look_at_from_position in my hand-rolled code quite a bit and was trying to do the same with Phantom Cam. The other thing I could think of would be z-axis flip. I need to use the use_model_front parameter in some places, due to the orientation of my model. This might explain the 180° flip.
